language-support/scala/codegen-testing/src/test/scala/com/digitalasset/ledger/client/binding/ValueGen.scala

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.client.binding
+
+import encoding.{ValuePrimitiveEncoding, GenEncoding}
+import org.scalacheck.{Arbitrary, Gen, Shrink}
+
+import scala.language.higherKinds
+
+object ValueGen {
```

Comment:
```suggestion
private[binding] object ValueGen {
```

`object ValueSpec` was public because `class ValueSpec` must be. This was just superstition on my part, I think.